internal/parse/parser.go

```diff
@@ -256,6 +263,25 @@ func (p *Parser) parseIdentifier() (string, bool) {
 	return "", false
 }
 
+// parseColumn parses a column made up of name bytes, optionally dot-prefixed by
+// its table name. parseColumn returns an error so that it can be used with
+// parseList.
+func (p *Parser) parseColumn() (FullName, bool, error) {
+	cp := p.save()
+	if id, ok := p.parseIdentifierAsterisk(); ok {
+		if id != "*" && p.skipByte('.') {
+			if idCol, ok := p.parseIdentifierAsterisk(); ok {
+				return FullName{Prefix: id, Name: idCol}, true, nil
+			}
+		} else {
+			// A column name specified without a table prefix should be in Name.
+			return FullName{Name: id}, true, nil
```

Comment:
'Should' sounds unsure, also I'm not sure this comment is necessary?

Reply:
Also meant to be part of a different PR.